airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_insight_streams.py

```diff
@@ -115,18 +117,24 @@ def state(self) -> MutableMapping[str, Any]:
             return {
                 self.cursor_field: self._cursor_value.isoformat(),
                 "slices": [d.isoformat() for d in self._completed_slices],
+                "time_increment": self.time_increment,
             }
 
         if self._completed_slices:
             return {
                 "slices": [d.isoformat() for d in self._completed_slices],
+                "time_increment": self.time_increment,
             }
 
         return {}
 
     @state.setter
     def state(self, value: Mapping[str, Any]):
-        """State setter"""
+        """State setter, will ignore saved state if time_increment is different from previous."""
+        if value.get("time_increment", 1) != self.time_increment:
```

Comment:
could you add some context on why we skip? not setting state at all seems like it would contradict the user/developer's expectations no?

Reply:
it is not, as we discussed earlier in other tickets #2154 if user changes connector settings its state is not valid in most of the cases, this is not supported on Airbyte platform level (UX/Core) so we do this at least on connector level.